cmake/tribits.cmake

```diff
@@ -95,10 +95,10 @@ ENDMACRO()
 
 # Older versions of cmake does not make include directories transitive
 MACRO(TARGET_LINK_AND_INCLUDE_LIBRARIES TARGET_NAME)
-  TARGET_LINK_LIBRARIES(${TARGET_NAME} LINK_PUBLIC ${ARGN})
+  TARGET_LINK_LIBRARIES(${TARGET_NAME} INTERFACE ${ARGN})
```

Comment:
The `INTERFACE` is only supported by cmake >= 3. This should be wrapped by an if statement, like what is done for `ADD_INTERFACE_LIBRARY`. Or perhaps kokkos could require cmake 3 and then drop the if statements altogether.